code/modules/events/crystal_event.dm

```diff
@@ -0,0 +1,477 @@
+	///If not active the crystal will not emit radiations and gases
+	var/active = TRUE
+	///Check if the crew managed to stop the ZK-class event
+	var/is_zk = FALSE
+
+/obj/machinery/destabilized_crystal/Initialize()
+	. = ..()
+	GLOB.destabilized_crystals += src
+
+/obj/machinery/destabilized_crystal/Destroy()
+	GLOB.destabilized_crystals -= src
+	if(is_zk)
+		priority_announce("WARNING - Portal are appearing everywhere, you failed to contain the event. You people should feel ashamed of yourself!","Alarm")
+	return..()
+
+/obj/machinery/destabilized_crystal/process()
+	if(active)
+		if(prob(75))
+			src.fire_nuclear_particle()
+			radiation_pulse(src, 250, 6)
+		var/turf/T = loc
+		var/datum/gas_mixture/env = T.return_air()
+		var/datum/gas_mixture/removed
+		var/gasefficency = 0.5
+		removed = env.remove(gasefficency * env.total_moles())
+		removed.assert_gases(/datum/gas/bz, /datum/gas/miasma)
+		removed.gases[/datum/gas/bz][MOLES] += 15.5
+		removed.gases[/datum/gas/miasma][MOLES] += 5.5
+		env.merge(removed)
+		air_update_turf()
+
+/obj/machinery/destabilized_crystal/proc/zk_event()
+	active = FALSE
+	priority_announce("WARNING - The crystal has reached critical instability point. ZK-Event inbound, please do not panic, anyone who panics will \
+						be terminated on the spot. Have a nice day", "Alert")
+	sound_to_playing_players('sound/machines/alarm.ogg')
+	sleep(15 SECONDS)
+
+	var/list/spawners = list()
+	for(var/obj/effect/landmark/event_spawn/temp in GLOB.generic_event_spawns)
+		if(QDELETED(temp))
+			continue
+		if(is_station_level(temp.loc.z))
+			spawners += temp
+
+	if(!spawners.len)
+		message_admins("No landmarks on the station, aborting")
+		return MAP_ERROR
+
+	var/obj/spawner = pick_n_take(spawners)
+	var/pick_portal = pickweight(GLOB.waves["huge wave"])
+	for(var/i = 0, i < 10, i++)
+		new pick_portal(spawner.loc)
+	explosion(src, 7, 10, 25, 25)
+	is_zk = TRUE
+	qdel(src)
+
+/obj/machinery/destabilized_crystal/attackby(obj/item/W, mob/living/user, params)
+	if(!istype(W) || (W.item_flags & ABSTRACT) || !istype(user))
+		return
+	if(istype(W, /obj/item/crystal_stabilizer))
+		var/obj/item/crystal_stabilizer/injector = W
+		if(!injector.filled)
+			to_chat(user, "<span class='notice'>\The [W] is empty!</span>")
+			return
+		to_chat(user, "<span class='notice'>You carefully begin inject \the [src] with \the [W]... please don't move untill all the steps are finished</span>")
+		if(W.use_tool(src, user, 5 SECONDS, volume=100))
+			to_chat(user, "<span class='notice'>Seems that \the [src] internal resonance is fading with the fluid!</span>")
+			playsound(get_turf(src), 'sound/effects/supermatter.ogg', 35, TRUE)
+			if(W.use_tool(src, user, 6.5 SECONDS, volume=100))
+				to_chat(user, "<span class='notice'>The [src] is reacting violently with the fluid!</span>")
+				src.fire_nuclear_particle()
+				radiation_pulse(src, 2500, 6)
+				if(W.use_tool(src, user, 7.5 SECONDS, volume=100))
```

Comment:
Please turn these into early returns.

Reply:
uh how? is a three processes one after another, i don't see a way of making it early return